src/Pos/Crypto/Arbitrary/Unsafe.hs

```diff
@@ -18,13 +20,13 @@ import           Pos.Crypto.SignTag        (SignTag)
 import           Pos.Util.Arbitrary        (ArbitraryUnsafe (..), arbitrarySizedSL)
 
 instance Bi PublicKey => ArbitraryUnsafe PublicKey where
-    arbitraryUnsafe = Bi.decode <$> arbitrarySizedSL 32
+    arbitraryUnsafe = Bi.decodeOrFail . BSL.toStrict <$> arbitrarySizedSL 32
```

Comment:
Perhaps it would be better to change `arbitrarySizedSL` to something that returns strict BS.